gix-hash/src/oid.rs

```diff
@@ -140,12 +152,32 @@ impl oid {
         }
     }
 
+    /// Write ourselves to the `out` in hexadecimal notation, returning the hex-string ready for display.
+    ///
+    /// **Panics** if the buffer isn't big enough to hold twice as many bytes as the current binary size.
+    #[inline]
+    #[must_use]
+    pub fn hex_to_buf<'a>(&self, buf: &'a mut [u8]) -> &'a mut str {
+        let num_hex_bytes = self.bytes.len() * 2;
+        faster_hex::hex_encode(&self.bytes, &mut buf[..num_hex_bytes]).expect("to count correctly")
+    }
+
+    /// Write ourselves to `out` in hexadecimal notation.
+    #[inline]
+    pub fn write_hex_to(&self, out: &mut dyn std::io::Write) -> std::io::Result<()> {
```

Comment:
Yeah, why not.
These methods feel oddly familiar, even though they are new.

Ah, now I see they just moved.